### PR TITLE
added faceUV option in CreateBox

### PR DESCRIPTION
**Pull request by jbousquie**
The ability to set a different UV per face
Exemple : face[1] has a different UV than the default one (0, 0, 1, 1)
```javascript
var faceUV = new Array(6);
faceUV[1] = new BABYLON.Vector4(0.5, 0.5, 0.6, 0.6);
  var options = {
    width: 10,
    height: 3,
    depth: 5,
    faceUV:  faceUV
  };
  var box = BABYLON.Mesh.CreateBox('box', options, scene);
```
Easy to set a different image per face with one line if the texture is an atlas !

Note : my branch has been mysteriously merged in my master, don't understand why ...

### CODE REVIEW

**Comment by jbousquie** [comment]
Obviously, it works as usual if we don't pass any faceUV parameter ...

File: src/Mesh/babylon.mesh.vertexData.ts [hunk 1]
```diff
@@ -574,7 +574,7 @@
             return vertexData;
         }
 
-        public static CreateBox(options: { width?: number, height?: number, depth?: number, sideOrientation?: number }): VertexData;
+        public static CreateBox(options: { width?: number, height?: number, depth?: number, faceUV: Vector4[], sideOrientation?: number }): VertexData;
```
Comment (deltakosh) [inline src/Mesh/babylon.mesh.vertexData.ts]: please mark it as optional (with ?)

File: src/Mesh/babylon.mesh.vertexData.ts [hunk 2]
```diff
@@ -594,6 +594,7 @@
             var width = 1;
             var height = 1;
             var depth = 1;
+            var faceUV = options.faceUV || new Array(6);
```
Comment (deltakosh) [inline src/Mesh/babylon.mesh.vertexData.ts]: please mark it as Array<Vector4>

Comment (jbousquie) [inline src/Mesh/babylon.mesh.vertexData.ts]: oki
I'll do both marks tomorrow when I'm back on my computer, unless you prefer to do it right now.

Comment (deltakosh) [inline src/Mesh/babylon.mesh.vertexData.ts]: no rush my friend:)

**Comment by jbousquie** [comment]
just added your recommandations

**Pull request merged**